Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -0,0 +1,250 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file KrBoxClusterFinder3D.cpp
+/// \brief Class source code for Krypton and X-ray events
+/// \author Philip Hauer <hauer@hiskp.uni-bonn.de>
+
+#include "TPCReconstruction/KrBoxClusterFinder.h"
+#include <iostream>
+
+using namespace o2::tpc;
+
+// Constructor:
+// This function creates a three dimensional vector (Pad,Row,Time) which is
+// filled with the charges
+KrBoxClusterFinder::KrBoxClusterFinder(std::vector<o2::tpc::Digit>* eventSector)
+{
+  std::vector<o2::tpc::Digit>::iterator cl = eventSector->begin();
+  std::vector<int> Rows, Pads, Times;
+
+  if (eventSector->size() == 0) {
+    // prevents a segementation fault if the envent contains no data
+    // segementation fault would occure later when trying to dereference the
+    // max/min pointer (which are nullptr if data is empty) empty events should
+    // be catched in the main function, hence, this "if" is no longer necessary
+    std::cout << "WARNING SECTOR SIZE (AMOUNT OF DATA POINTS) IN CURRENT RUN IS 0" << std::endl;
+    std::cout << "mMapOfAllDigits WITH 0's IS GENERATED" << std::endl;
+
+    mMapOfAllDigits = std::vector<std::vector<std::vector<int>>>(10, std::vector<std::vector<int>>(10, std::vector<int>(10, 0)));
+  } else {
+    for (int i = 0; (cl + i) < eventSector->end(); i++) {
+      Times.push_back((cl + i)->getTimeStamp());
+      Pads.push_back((cl + i)->getPad());
+      Rows.push_back((cl + i)->getRow());
+    }
+    // here a "mMapOfAllDigits" large enough to contain all data points of the envents is
+    // initialized and the filled
+    mMapOfAllDigits = std::vector<std::vector<std::vector<int>>>(mMaxTimes, std::vector<std::vector<int>>(mMaxRows, std::vector<int>(mMaxPads)));
+    for (int i = 0; (cl + i) < eventSector->end(); i++) {
+      mMapOfAllDigits.at(Times.at(i)).at(Rows.at(i)).at(Pads.at(i)) = (cl + i)->getChargeFloat();

```

Comment:
If you don't need range checks (you are confident that you are always inside the valid limits), it is better to use `[]` for access since it is faster.

Reply:
Done.